### PR TITLE
If the product has no combinations then the `specific_references` must be filled in

### LINKED ISSUE #15783: ISBN, EAN13, UPC and MPN are not displayed in FO for standard products without combinations
**Describe the bug**
ISBN, EAN13, UPC & MPN are not displayed in FO for standard products

**To Reproduce**
1. Go to Catalog > Products and edit a standard product without combinations
2. Click on Options tab
3. Fill ISBN, EAN13, UPC & MPN fields
4. Go to product page in FO & click on product details tab
5. ISBN, EAN13, UPC & MPN are not displayed while they should be. It's working for a product with combinations.
If these fields are not filled for the selected combination, then you see what is filled in options tab. If fields are filled for the combinations, you should see what is configured for the combinations.

(MPN was added by this PR : #12855) 

**Standard product NOK**

capture d'écran_2038

capture d'écran_2039

**Product with combinations OK**

capture d'écran_2041

capture d'écran_2040

capture d'écran_2043

capture d'écran_2042

**Additional information**
PrestaShop version: 1.7.6.1
PHP version: N/A

### PR DESCRIPTION
**Pull request by levyn**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | #15783
| Type?             | bug fix
| Category?         | FO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #15783
| How to test?      | #15783
| Possible impacts? | #15783

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/24792)

### CODE REVIEW

**Review by matthieu-rolland — COMMENTED** [review]
Apparently, CS fixer is not happy, could you run this command from the root of your prestashop, and then commit the modification?

`./vendor/friendsofphp/php-cs-fixer/php-cs-fixer fix src/Adapter/Presenter/Product/ProductLazyArray.php`

File: src/Adapter/Presenter/Product/ProductLazyArray.php [hunk 1]
```diff
@@ -542,6 +542,22 @@ public function getMainVariants()
...
+                $this->product['attributes'][0]['upc'] = $this->product['upc'];
+            }
+            if (isset($this->product['ean13'])) {
+                $this->product['attributes'][0]['ean13'] = $this->product['ean13'];
+            }
+            if (isset($this->product['mpn'])) {
+                $this->product['attributes'][0]['mpn'] = $this->product['mpn'];
+            }
+        }
+
```
Comment (kpodemski) [inline src/Adapter/Presenter/Product/ProductLazyArray.php]:
```suggestion
        if (!isset($this->product['attributes'])) {
            $this->product['attributes'][0]['isbn'] = $this->product['isbn'] ?? false;
            $this->product['attributes'][0]['upc'] = $this->product['upc'] ?? false;
            $this->product['attributes'][0]['ean13'] = $this->product['ean13'] ?? false;
            $this->product['attributes'][0]['mpn'] = $this->product['mpn'] ?? false;
        }

```

**Review by kpodemski — CHANGES_REQUESTED** [review]
I'd recommend doing it a bit simpler

File: src/Adapter/Presenter/Product/ProductLazyArray.php (hunk 1 above)
Comment (levyn) [reply src/Adapter/Presenter/Product/ProductLazyArray.php]: Thanks @kpodemski

**Review by PierreRambaud — CHANGES_REQUESTED** [review]
I'm not sure it's the right way to do it. A production without combinations shouldn't have to take care of attributes as data are already in the `->product` array.
It looks more like a template problem.

**Comment by kpodemski** [comment]
@PierreRambaud you're right about `$this->product['attributes']` **but** I think having it fixed in this place would be better because this automatically fixes all themes without touching them - maybe we shouldn't base on `$this->product['attributes']` but instead use temporary array

File: src/Adapter/Presenter/Product/ProductLazyArray.php [hunk 2]
```diff
@@ -538,10 +538,18 @@ public function getMainVariants()
...
+        // If the product has no combinations then the `specific_references` must be filled in
+        if (!isset($this->product['attributes'])) {
+            $this->product['attributes'][0]['isbn'] = $this->product['isbn'] ?? false;
+            $this->product['attributes'][0]['upc'] = $this->product['upc'] ?? false;
+            $this->product['attributes'][0]['ean13'] = $this->product['ean13'] ?? false;
+            $this->product['attributes'][0]['mpn'] = $this->product['mpn'] ?? false;
+        }
+
         if (isset($this->product['attributes']) && !isset($this->product['cart_quantity'])) {
             $specificReferences = array_slice($this->product['attributes'], 0)[0];
```
Comment (PierreRambaud) [inline src/Adapter/Presenter/Product/ProductLazyArray.php]:
```suggestion
        if (!isset($this->product['cart_quantity'])) {
            return null;
        }
        
        // If the product has no combinations then the `specific_references` must be filled in
        if (isset($this->product['attributes'])) {
            $specificReferences = array_slice($this->product['attributes'], 0)[0];
        } else {
            $specificReferences = [
                'isbn' => $this->product['isbn'] ?? false,
                'upc' => $this->product['upc'] ?? false,
                'ean13' => $this->product['ean13'] ?? false,
                'mpn' => $this->product['mpn'] ?? false,
            ];
        }           
```

@kpodemski Maybe something like that will be better for everyone?

Comment (kpodemski) [reply src/Adapter/Presenter/Product/ProductLazyArray.php]: @PierreRambaud yeah, looking good, can you move from there @levyn?

Comment (levyn) [reply src/Adapter/Presenter/Product/ProductLazyArray.php]: @kpodemski Yes, I like it. Just a little mistake, "!" - it is not necessary here

**Review by PierreRambaud — APPROVED** [review]

**Review by kpodemski — APPROVED** [review]

**Comment by florine2623** [comment]
LGTM

Thanks @levyn !

**Comment by Progi1984** [comment]
Thanks @levyn & @florine2623

**Pull request merged**